### DISCUSSION TITLE
Feature Request: python: Allow passing pathlib.Path to `connect()`

### DISCUSSION THREAD

**Discussion by NickCrews** [duckdb/duckdb#6447] [Ideas]
Currently (on master) you can only pass the location of the database file as a string. It's not THAT big of a deal for me as a user to go `connect(str(my_path))`, but I'd imagine it wouldn't be that hard to implement? Thank you!

**Comment by Tishj** [comment]
Yes definitely something we need to add!

I'll see if we can add an implicit cast in pybind to convert a `pathlib.Path` instance to std::string

https://pybind11.readthedocs.io/en/stable/advanced/cast/custom.html

Sounds like it's possible ;)
That would allow you to pass a pathlib.Path instance to functions that expect a string

**Reply by NickCrews** [reply]
I would offer to write this as a PR, but that stuff with pybind looks too advanced for me. If there's anything else I can do to usher this along, let me know!

**Reply by Tishj** [reply]
I've worked on this, but I hit a bit of a roadblock, due to the fact that pybind11 already does a lot of implicit casting to `std::string` I couldn't find a way to *add* to this, only completely override it with my own casting - which would mean having to rewrite all of the implicit casting they perform.

**Reply by NickCrews** [reply]
Gotcha. IDK if this is possible, but could we have a thin pure python layer that does the casting Path -> str, and then that layer calls down into the C generated from pybind, so we don't have to do any casting inside C++/pybind? If we can cherry-pick which methods have this wrapper layer then that seems worth it. If we have to duplicate the entire API in this python layer, then that is more overhead than is worth it.

**Reply by Tishj** [reply]
I think I know how we can get this working without resorting to native python - I was just too hardheaded in wanting a drop-in implicit conversion solution.
I'll try to write something up in the coming days

**Reply by Tishj** [reply]
PR is here #6835
